src/zenml/utils/secret_utils.py

```diff
@@ -107,25 +114,65 @@ def ClearTextField(*args: Any, **kwargs: Any) -> Any:
     return Field(*args, **kwargs)  # type: ignore[pydantic-field]
 
 
-def is_secret_field(field: "ModelField") -> bool:
+def is_secret_field(field: "FieldInfo") -> bool:
     """Returns whether a pydantic field contains sensitive information or not.
 
     Args:
         field: The field to check.
 
+    Raises:
+        ValueError: if the extras for the field are set incorrectly.
+
     Returns:
         `True` if the field contains sensitive information, `False` otherwise.
     """
-    return field.field_info.extra.get(PYDANTIC_SENSITIVE_FIELD_MARKER, False)  # type: ignore[no-any-return]
-
-
-def is_clear_text_field(field: "ModelField") -> bool:
+    if field.json_schema_extra is not None:
+        if isinstance(field.json_schema_extra, dict):
+            if marker := field.json_schema_extra.get(
+                PYDANTIC_SENSITIVE_FIELD_MARKER
+            ):
+                assert isinstance(marker, bool), (
+                    f"The parameter `{PYDANTIC_SENSITIVE_FIELD_MARKER}` in the "
+                    f"field definition can only be a boolean value."
+                )
+                return marker
+
+        else:
+            raise ValueError(
```

Comment:
Can we not just return `False` here? I'm not sure what it means that the extras are set incorrectly, but I assume if the field is a secret field (which means it's coming from our code) it will be set correcty?

Same applies to the `is_clear_text_field` function below

Reply:
I wanted to raise a `ValueError` here because of a case that I was not able to fully understand myself. To explain it quickly, you can check the following code: 

```python
from pydantic import BaseModel, Field
from zenml.utils.secret_utils import SecretField


class A(BaseModel):
    a: int = Field(default=3, random_parameter=5)


a_json_schema_extra = A().model_fields["a"].json_schema_extra
print(a_json_schema_extra)  # -> {'random_parameter': 5}
print(type(a_json_schema_extra))  # -> <class 'dict'>


class B(BaseModel):
    b: int = SecretField(default=3, random_parameter=5)


b_json_schema_extra = B().model_fields["b"].json_schema_extra
print(b_json_schema_extra)  # -> {'random_parameter': 5, 'sensitive': True}
print(type(b_json_schema_extra))  # -> <class 'dict'>


class C(BaseModel):
    c: int = Field(default=3)


c_json_schema_extra = C().model_fields["c"].json_schema_extra
print(c_json_schema_extra)  # -> None
```

So, when you define a pydantic `Field`, you can put extra values there which now directly go into this `json_schema_extra` value. However, if you check the source code of `FieldInfo`, you will see:


```python
class FieldInfo(_repr.Representation):
    """This class holds information about a field.
        ... 
        json_schema_extra: A dict or callable to provide extra JSON schema properties.
        ...
    """
    ...
    json_schema_extra: JsonDict | typing.Callable[[JsonDict], None] | None
    ...
```

As you can see from my code, I can find cases where we set it `None` or a `JsonDict`, however, I can not find any proper cases where it can become this `Callable`, which takes a `JsonDict` as input and returns `None`. Keep in mind, that the `SecretField` can also be used by our users when they define their own flavors (based on our flavor implementations). Just to be on the safe side of this, I wanted to raise a `ValueError`.

Reply:
In your current implementation, both this method and the `is_clear_text_field` fail, which means even if people define such a valid (in pydantic terms) field, ZenML would always crash for them.

My suggestions: `False` here, `True` for clear text field

Reply:
For more info here, the current implementations are working as implemented. However, your suggestion here is also valid. I have changed the function accordingly. It now throws a warning instead of failing if the `json_schema_extra` is not a `dict`. Since both secret and clear text fields need to be explicitly defined, I am returning `False` on default on both.